### PR TITLE
Clean up docker-api

### PR DESCRIPTION
**Pull request by freva**

### CODE REVIEW

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/docker/DockerOperationsImpl.java [hunk 1]
```diff
@@ -323,7 +300,12 @@ public void stopServicesOnNode(ContainerName containerName) {
     }
 
     @Override
-    public Optional<Docker.ContainerStats> getContainerStats(ContainerName containerName) {
+    public void trySuspendNode(ContainerName containerName) {
+        executeCommandInContainer(containerName, nodeProgram, "suspend");
```
Comment (hakonhall) [inline node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/docker/DockerOperationsImpl.java]: This changes suspend failures from non-fatal to fatal, why?

Comment (freva) [reply node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/docker/DockerOperationsImpl.java]: I wasn't sure what to do with this one, none of the other nodectl commands are handled this way. I added checking for when container is missing the only place this is called: https://github.com/vespa-engine/vespa/pull/6998/files#diff-22e52ac72910bf612ee17481b6d4d5c8R298
Additionally, there is a try/catch on Exception here: https://github.com/vespa-engine/vespa/pull/6998/files#diff-22e52ac72910bf612ee17481b6d4d5c8R341

**Comment by freva** [comment]
Build failure unrelated, java tests successful.

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/docker/DockerOperationsImpl.java (hunk 1 above)
Comment (hakonhall) [reply node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/docker/DockerOperationsImpl.java]: Stop services won't be called if suspend fails, with this change. That seems unwarranted. The suspension is just an optimization to make it faster to start again afterwards.

**Comment by freva** [comment]
@hakonhall PTAL. 

Changed back, but added re-throwing of `ContainerNotFoundException` - no point in try to stop services if suspend failed because container is gone.

**Review by hakonhall — APPROVED** [review]

**Pull request merged**